internal/core/services/vin_decoding_service.go

```diff
@@ -264,3 +289,30 @@ func buildFromDD(vin string, info *repoModel.DeviceDefinition) *models.VINDecodi
 
 	return v
 }
+
+func buildFromDATGroup(info *gateways.GetVehicleIdentificationByVinResponse) (*models.VINDecodingInfoData, error) {
+	raw, _ := xml.Marshal(info)
+	if info == nil {
+		return nil, fmt.Errorf("vin info was nil")
+	}
+
+	dossier := info.Body.GetDataVehicleIdentificationByVinResponse.VXS.Dossier[0].Vehicle
+	v := &models.VINDecodingInfoData{
+		VIN:  dossier.VINResult.VINVehicle.VINumber.VinCode,
+		Year: int32(dossier.BuildYear),
+		//Year:       2022,
+		Make:       strings.TrimSpace(dossier.ManufacturerName),
+		Model:      strings.TrimSpace(dossier.BaseModelName),
+		Source:     models.DATGroupProvider,
+		ExternalID: dossier.VINResult.VINVehicle.VINumber.VinCode,
+		//StyleName:  info.GetStyle(),
```

Comment:
we'll need to think of something for the style name

Reply:
What would be the fields to be able to map them?